src/libcollections/vec.rs

```diff
@@ -659,6 +659,43 @@ impl<T> Vec<T> {
         }
     }
 
+
+    /// Move all elements from one vector to another
+    ///
+    /// # Panics
+    ///
+    /// Panics if the number of elements in the vector overflows a `uint`.
+    ///
+    /// # Examples
+    /// ```rust
+    /// let mut vec = vec![1i, 2, 3];
+    /// let mut vec2 = vec![4i, 5, 6];
+    /// vec.append(&mut vec2);
+    /// assert_eq!(vec, vec![1i, 2, 3, 4, 5, 6]);
+    /// assert_eq!(vec2, Vec::new::<i>());
+    /// ```
+    #[inline]
+    #[stable]
```

Comment:
The should be left unstable since this a new API that we haven't let "bake" enough. Something like

`unstable = "new API, waiting for dust to settle"`